tests/chart_tests/test_elasticsearch.py

```diff
@@ -279,6 +279,22 @@ def test_elastic_nginx_config_pattern_defaults(self, kube_version):
                 "location ~ ^/ { deny all; } } }",
             ]
         )
+        assert "client_max_body_size 100M" in nginx_config
+
+    def test_elastic_nginx_config_custom_max_body_size(self, kube_version):
+        """Test that custom max body size is properly set."""
+        docs = render_chart(
+            kube_version=kube_version,
+            values={"elasticsearch": {"nginx": {"maxBodySize": "200M"}}},
+            show_only=[
+                "charts/elasticsearch/templates/nginx/nginx-es-configmap.yaml",
+            ],
+        )
+
+        assert len(docs) == 1
+        doc = docs[0]
+        nginx_config = " ".join(doc["data"]["nginx.conf"].split())
+        assert "client_max_body_size 200M" in nginx_config
```

Comment:
Let's make this something very conspicuous, something we would never actually choose, so it stands out as a test value instead of something that we might actually set it to some day.
```suggestion
            values={"elasticsearch": {"nginx": {"maxBodySize": "123456789M"}}},
            show_only=[
                "charts/elasticsearch/templates/nginx/nginx-es-configmap.yaml",
            ],
        )

        assert len(docs) == 1
        doc = docs[0]
        nginx_config = " ".join(doc["data"]["nginx.conf"].split())
        assert "client_max_body_size 123456789M" in nginx_config
```